compiler-plugin/src/main/kotlin/arrow/meta/quotes/MetaExtensions.kt

```diff
@@ -0,0 +1,271 @@
+package arrow.meta.quotes
+
+import arrow.meta.Meta
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.quotes.classorobject.ClassDeclaration
+import arrow.meta.quotes.classorobject.ObjectDeclaration
+import arrow.meta.quotes.declaration.DestructuringDeclaration
+import arrow.meta.quotes.element.CatchClause
+import arrow.meta.quotes.element.FinallySection
+import arrow.meta.quotes.element.ImportDirective
+import arrow.meta.quotes.element.ParameterList
+import arrow.meta.quotes.element.ValueArgument
+import arrow.meta.quotes.element.WhenCondition
+import arrow.meta.quotes.expression.AnnotatedExpression
+import arrow.meta.quotes.expression.BinaryExpression
+import arrow.meta.quotes.expression.BlockExpression
+import arrow.meta.quotes.expression.IfExpression
+import arrow.meta.quotes.expression.LambdaExpression
+import arrow.meta.quotes.expression.ThrowExpression
+import arrow.meta.quotes.expression.TryExpression
+import arrow.meta.quotes.expression.expressionwithlabel.ReturnExpression
+import arrow.meta.quotes.filebase.File
+import arrow.meta.quotes.modifierlist.ModifierList
+import arrow.meta.quotes.modifierlist.TypeReference
+import arrow.meta.quotes.nameddeclaration.notstubbed.FunctionLiteralScope
+import arrow.meta.quotes.nameddeclaration.stub.typeparameterlistowner.NamedFunction
+import arrow.meta.quotes.nameddeclaration.stub.typeparameterlistowner.Property
+import arrow.meta.quotes.nameddeclaration.stub.typeparameterlistowner.TypeAlias
+import org.jetbrains.kotlin.psi.KtAnnotatedExpression
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtCatchClause
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtDestructuringDeclaration
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtFinallySection
+import org.jetbrains.kotlin.psi.KtFunctionLiteral
+import org.jetbrains.kotlin.psi.KtIfExpression
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtLambdaExpression
+import org.jetbrains.kotlin.psi.KtModifierList
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+import org.jetbrains.kotlin.psi.KtParameterList
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtReturnExpression
+import org.jetbrains.kotlin.psi.KtThrowExpression
+import org.jetbrains.kotlin.psi.KtTryExpression
+import org.jetbrains.kotlin.psi.KtTypeAlias
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.psi.KtWhenCondition
+
+/**
+ * ```""" $`@annotations` $expression """.annotatedExpression```
```

Comment:
These should be all ank typechecked snippets or they can fall behind the API as it evolves

Reply:
I'm not sure how this can be run with typechecking - without being wrapped as a declaration within the Transform object, I'm afraid the code doesn't know what this is by itself.

I've marked this "blocked" for now as I'm inexperienced with ank, but I will continue to work on adding more examples in the meantime

Reply:
Update: unblocked by @franciscodr - we agreed to not worry about ank for this particular PR, but to merge in all the examples and the file structure as he looks more into formatting/ank stuff perhaps in another PR